File: docs-examples/pubsub-java/src/test/java/io/micronaut/configuration/lettuce/docs/pubsub/BookPublisher.java
package io.micronaut.configuration.lettuce.docs.pubsub;

import io.micronaut.configuration.lettuce.pubsub.RedisPubSubPublisher;
import jakarta.inject.Singleton;

@Singleton
class BookPublisher {
    private final RedisPubSubPublisher publisher;

    BookPublisher(RedisPubSubPublisher publisher) {
        this.publisher = publisher;
    }

    // tag::publisher[]
    void publish(BookCreated event) {
        publisher.publish("books.created", event);
    }
    // end::publisher[]
}

File: docs-examples/pubsub-kotlin/src/test/kotlin/io/micronaut/configuration/lettuce/docs/pubsub/BookPublisher.kt
package io.micronaut.configuration.lettuce.docs.pubsub

import io.micronaut.configuration.lettuce.pubsub.RedisPubSubPublisher
import jakarta.inject.Singleton

@Singleton
class BookPublisher(private val publisher: RedisPubSubPublisher) {

    // tag::publisher[]
    fun publish(event: BookCreated) {
        publisher.publish("books.created", event)
    }
    // end::publisher[]
}
